cloud/deployment/workerqueue/workerqueue_test.go

```diff
@@ -886,28 +886,36 @@ func TestSetWorkerQueueValues(t *testing.T) {
 		MinWorkerCount:    0,
 		WorkerConcurrency: 0,
 	}
+	mockMachineOptions := &astroplatformcore.WorkerMachine{
+		Name: "a5",
+		Concurrency: astroplatformcore.Range{
+			Default: 180,
+			Ceiling: 275,
+			Floor:   175,
+		},
+	}
 	t.Run("sets user provided min worker count for queue", func(t *testing.T) {
-		actualQueue := SetWorkerQueueValues(0, 150, 225, mockWorkerQueue, mockWorkerQueueDefaultOptions)
+		actualQueue := SetWorkerQueueValues(0, 150, 225, mockWorkerQueue, mockWorkerQueueDefaultOptions, mockMachineOptions)
 		assert.Equal(t, mockWorkerQueue.MinWorkerCount, actualQueue.MinWorkerCount)
 	})
 	t.Run("sets user provided max worker count for queue", func(t *testing.T) {
-		actualQueue := SetWorkerQueueValues(10, 150, 225, mockWorkerQueue, mockWorkerQueueDefaultOptions)
+		actualQueue := SetWorkerQueueValues(10, 150, 225, mockWorkerQueue, mockWorkerQueueDefaultOptions, mockMachineOptions)
 		assert.Equal(t, mockWorkerQueue.MaxWorkerCount, actualQueue.MaxWorkerCount)
 	})
 	t.Run("sets user provided worker concurrency for queue", func(t *testing.T) {
-		actualQueue := SetWorkerQueueValues(10, 150, 225, mockWorkerQueue, mockWorkerQueueDefaultOptions)
+		actualQueue := SetWorkerQueueValues(10, 150, 225, mockWorkerQueue, mockWorkerQueueDefaultOptions, mockMachineOptions)
 		assert.Equal(t, mockWorkerQueue.WorkerConcurrency, actualQueue.WorkerConcurrency)
 	})
 	t.Run("sets default min worker count for queue if user did not provide it", func(t *testing.T) {
-		actualQueue := SetWorkerQueueValues(-1, 150, 225, mockWorkerQueue, mockWorkerQueueDefaultOptions)
+		actualQueue := SetWorkerQueueValues(-1, 150, 225, mockWorkerQueue, mockWorkerQueueDefaultOptions, mockMachineOptions)
 		assert.Equal(t, int(mockWorkerQueueDefaultOptions.MinWorkers.Default), actualQueue.MinWorkerCount)
 	})
 	t.Run("sets default max worker count for queue if user did not provide it", func(t *testing.T) {
-		actualQueue := SetWorkerQueueValues(10, 0, 225, mockWorkerQueue, mockWorkerQueueDefaultOptions)
+		actualQueue := SetWorkerQueueValues(10, 0, 225, mockWorkerQueue, mockWorkerQueueDefaultOptions, mockMachineOptions)
 		assert.Equal(t, int(mockWorkerQueueDefaultOptions.MaxWorkers.Default), actualQueue.MaxWorkerCount)
 	})
 	t.Run("sets default worker concurrency for queue if user did not provide it", func(t *testing.T) {
-		actualQueue := SetWorkerQueueValues(10, 150, 0, mockWorkerQueue, mockWorkerQueueDefaultOptions)
+		actualQueue := SetWorkerQueueValues(10, 150, 0, mockWorkerQueue, mockWorkerQueueDefaultOptions, mockMachineOptions)
```

Comment:
for this test are `workerQueueDefaultOptions.WorkerConcurrency.Default` and `machineOptions.Concurrency.Default` the same value so it wasn't caught?

Reply:
I don't think if they were different values it would be caught. `workerQueueDefaultOptions.WorkerConcurrency.Default` is only used by hybrid and `machineOptions.Concurrency.Default`. But SetWorkerQueueValues was setup assuming that they both use `workerQueueDefaultOptions.WorkerConcurrency.Default` and the tests were setup to reflect that. Just need to make sure that SetWorkerQueueValues uses mockMachineOptions for hosted deployments which the tests do now